### PR TITLE
Fixed unsafe casting

### PR DESCRIPTION
**Pull request by SanoopC**
ClassCastException is fixed by checking the exception type

### CODE REVIEW

File: sift/src/main/java/siftscience/android/AppStateCollector.java [hunk 1]
```diff
@@ -284,17 +284,22 @@ public void onSuccess(LocationSettingsResponse locationSettingsResponse) {
...
+                                        break;
+                                    case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
+                                        Log.i(TAG, "Location settings are inadequate, and cannot be " +
+                                                "fixed here. Fix in Settings.");
+                                }
+                            } else {
+                                // A different, unknown type of error occurred.
+                                Log.d(TAG, "ERROR! " + e.getMessage());
                             }
+                            mRequestingLocationUpdates = false;
```
Comment (vincent-sift) [inline sift/src/main/java/siftscience/android/AppStateCollector.java:302]: Shouldn't this line be moved to line 297, to match the same logic as the previous version?

File: sift/src/main/java/siftscience/android/AppStateCollector.java [hunk 2]
```diff
@@ -284,17 +284,22 @@ public void onSuccess(LocationSettingsResponse locationSettingsResponse) {
...
+                                        Log.i(TAG, "Location settings are not satisfied. " +
+                                                "Try to attempt upgrade location settings");
+                                        break;
+                                    case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
+                                        Log.i(TAG, "Location settings are inadequate, and cannot be " +
+                                                "fixed here. Fix in Settings.");
+                                }
+                            } else {
+                                // A different, unknown type of error occurred.
+                                Log.d(TAG, "ERROR! " + e.getMessage());
```
Comment (vincent-sift) [inline sift/src/main/java/siftscience/android/AppStateCollector.java:300]: Please change this to 
```suggestion
                                Log.d(TAG, "Encountered unknown exception in checkLocationSettings", e);
```

**Review by vincent-sift — CHANGES_REQUESTED** [review]

File: sift/src/main/java/siftscience/android/AppStateCollector.java (hunk 1 above)
Comment (SanoopC) [reply sift/src/main/java/siftscience/android/AppStateCollector.java:302]: In both cases that line is required, that is why it was moved outside.

Comment (vincent-sift) [reply sift/src/main/java/siftscience/android/AppStateCollector.java:302]: With the old logic I believe that mRequestingLocationUpdates would only equal false when the statusCode = LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE and not in both cases, right?
```suggestion
switch (statusCode) {
    case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
        Log.i(TAG, "Location settings are not satisfied. " +
                "Try to attempt upgrade location settings");
        break;
    case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
        Log.i(TAG, "Location settings are inadequate, and cannot be " +
                "fixed here. Fix in Settings.");
        mRequestingLocationUpdates = false;
```

**Review by vincent-sift — APPROVED** [review]
LGTM

**Pull request merged**